src/server.c

```diff
@@ -4592,6 +4586,13 @@ void populateCommandTable(void) {
         if (populateSingleCommand(c,c->sflags) == C_ERR)
             serverPanic("Unsupported command flag or key spec flag");
 
+
+        if (!(c->flags & CMD_SENTINEL) && server.sentinel_mode)
+            continue;
+
+        if (c->flags & CMD_ONLY_SENTINEL && !server.sentinel_mode)
+            continue;
+
         retval1 = dictAdd(server.commands, sdsnew(c->name), c);
```

Comment:
Should this piece of code also be moved down to this position, if `continue` it should not insert a record into `commandId`.
````
/* Assign the ID used for ACL. */
c->id = ACLGetCommandID(c->name);
```